### ISSUE TITLE
FlorisKeyboard Crashed when press on second alternative words on RTL languages

### ISSUE THREAD

**Issue by kurdikeyboard** [florisboard/florisboard#199]
FlorisBoard version: 0.3.2

Source: F-Droid

Device: Xiaomi Mi 9T Pro

Rom: MiUi 12.0.3 Stable with Android 10 (Not Rooted)

This crash happened when i tried to press the hint words on the main word (ف) on Arabic keyboard i tried to write Kurdish Veh (ڤ) then suddenly stopped working!

Here is the error report

~~~ 1610919166919.stacktrace ~~~

java.lang.IndexOutOfBoundsException: Index: -1, Size: 5
	at androidx.core.view.ViewGroupKt.get(ViewGroup.kt:32)
	at dev.patrickgold.florisboard.ime.popup.KeyPopupManager.getActiveKeyData(KeyPopupManager.kt:493)
	at dev.patrickgold.florisboard.ime.text.key.KeyView.onFlorisTouchEvent(KeyView.kt:303)
	at dev.patrickgold.florisboard.ime.text.keyboard.KeyboardView.sendFlorisTouchEvent(KeyboardView.kt:206)
	at dev.patrickgold.florisboard.ime.text.keyboard.KeyboardView.onTouchEvent(KeyboardView.kt:184)
	at android.view.View.dispatchTouchEvent(View.java:13551)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3082)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2767)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3088)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2781)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3088)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2781)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3088)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2781)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3088)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2781)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3088)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2781)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3088)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2781)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3088)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2781)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3088)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2781)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3088)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2781)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3088)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2781)
	at com.android.internal.policy.DecorView.superDispatchTouchEvent(DecorView.java:497)
	at com.android.internal.policy.PhoneWindow.superDispatchTouchEvent(PhoneWindow.java:1853)
	at android.app.Dialog.dispatchTouchEvent(Dialog.java:863)
	at android.inputmethodservice.SoftInputWindow.dispatchTouchEvent(SoftInputWindow.java:161)
	at com.android.internal.policy.DecorView.dispatchTouchEvent(DecorView.java:455)
	at android.view.View.dispatchPointerEvent(View.java:13812)
	at android.view.ViewRootImpl$ViewPostImeInputStage.processPointerEvent(ViewRootImpl.java:5649)
	at android.view.ViewRootImpl$ViewPostImeInputStage.onProcess(ViewRootImpl.java:5449)
	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4950)
	at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:5003)
	at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:4969)
	at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:5109)
	at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:4977)
	at android.view.ViewRootImpl$AsyncInputStage.apply(ViewRootImpl.java:5166)
	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4950)
	at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:5003)
	at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:4969)
	at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:4977)
	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4950)
	at android.view.ViewRootImpl.deliverInputEvent(ViewRootImpl.java:7702)
	at android.view.ViewRootImpl.doProcessInputEvents(ViewRootImpl.java:7671)
	at android.view.ViewRootImpl.enqueueInputEvent(ViewRootImpl.java:7632)
	at android.view.ViewRootImpl$WindowInputEventReceiver.onInputEvent(ViewRootImpl.java:7832)
	at android.view.InputEventReceiver.dispatchInputEvent(InputEventReceiver.java:251)
	at android.os.MessageQueue.nativePollOnce(Native Method)
	at android.os.MessageQueue.next(MessageQueue.java:336)
	at android.os.Looper.loop(Looper.java:181)
	at android.app.ActivityThread.main(ActivityThread.java:7560)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:539)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950)

**Comment by patrickgold** [comment]
Thanks for your report! I've just released v0.3.3, which should arrive at your device in a few days. v0.3.3 completely reworks the extended popups, so I cannot say 100% that it fixed this issue, but there's a significant chance that it did. Could you please give me feedback when v0.3.3 arrives at your device if this fixed this crash? This would be awesome, thank you!

**Comment by kurdikeyboard** [comment]
> Thanks for your report! I've just released v0.3.3, which should arrive at your device in a few days. v0.3.3 completely reworks the extended popups, so I cannot say 100% that it fixed this issue, but there's a significant chance that it did. Could you please give me feedback when v0.3.3 arrives at your device if this fixed this crash? This would be awesome, thank you!

Im using the latest 0.3.3 i think the problem have been fixed i tried many times to write Hint text but it works great Thank you for your hard work you are amazing

**Comment by patrickgold** [comment]
Perfect thank you for confirming! I will close this issue now as it seems to have been resolved.
